components/scream/src/physics/rrtmgp/atmosphere_radiation.cpp

```diff
@@ -1,52 +1,272 @@
 #include "physics/rrtmgp/atmosphere_radiation.hpp"
+#include "physics/rrtmgp/rrtmgp_inputs_initializer.hpp"
+#include "mo_gas_concentrations.h"
 
 namespace scream {
-    RRTMGPRadiation::RRTMGPRadiation (const ekat::Comm& comm, const ekat::ParameterList& params) : m_rrtmgp_comm (comm), m_rrtmgp_params (params) {
+    RRTMGPRadiation::RRTMGPRadiation (const ekat::Comm& comm, const ekat::ParameterList& params) 
+        : AtmosphereProcess::AtmosphereProcess(), m_rrtmgp_comm (comm), m_rrtmgp_params (params) {
         /*
          * Anything that can be initialized without grid information can be initialized here.
          * I.e., universal constants, options, etc.
          */
+          if (!m_rrtmgp_params.isParameter("Grid")) {
+              m_rrtmgp_params.set("Grid",std::string("SE Physics"));
+          }
+
+          m_initializer = create_field_initializer<RRTMGPInputsInitializer>();
     }  // RRTMGPRadiation::RRTMGPRadiation
 
     void RRTMGPRadiation::set_grids(const std::shared_ptr<const GridsManager> grids_manager) {
 
+        using namespace ekat::units;
+
+        auto kgkg = kg/kg;
+        kgkg.set_string("kg/kg");
+        auto m3 = m * m * m;
+        m3.set_string("m3");
+        auto Wm2 = W / m / m;
+        Wm2.set_string("Wm2");
+        auto nondim = m/m;  // dummy unit for non-dimensional fields
+        auto micron = m / 1000000;
+
+        auto VL  = FieldTag::VerticalLevel;
+        auto COL = FieldTag::Column;
+        auto VAR = FieldTag::Variable;
+        auto CMP = FieldTag::Component;
+        constexpr int NVL = 42;  /* TODO THIS NEEDS TO BE CHANGED TO A CONFIGURABLE */
+
+        auto grid = grids_manager->get_grid("Physics");
+        const int ncol = grid->get_num_local_dofs();
+
+        // Need to hard-code some dimension sizes for now. TODO: find a better way of configuring this
+        int nswbands = 14;
+        int nlwbands = 16;
+        int ngas = 8;
+
+        // Set up dimension layouts
+        FieldLayout scalar2d_layout     { {COL   }, {ncol    } };
+        FieldLayout scalar3d_layout_mid { {COL,VL}, {ncol,NVL} };
+        FieldLayout scalar3d_layout_int { {COL,VL}, {ncol,NVL+1} };
+        // Use VAR field tag for gases for now; consider adding a tag?
+        FieldLayout gas_layout          { {VAR,COL,VL}, {ngas,ncol,NVL} };
+        FieldLayout scalar2d_swband_layout { {CMP,COL}, {nswbands,ncol} };
+
+        // Set required (input) fields here
+        m_required_fields.emplace("pmid" , scalar3d_layout_mid, Pa, grid->name());
+        m_required_fields.emplace("pint", scalar3d_layout_int, Pa, grid->name());
+        m_required_fields.emplace("tmid" , scalar3d_layout_mid, K , grid->name());
+        m_required_fields.emplace("tint" , scalar3d_layout_int, K , grid->name());
+        m_required_fields.emplace("col_dry", scalar3d_layout_mid, kgkg, grid->name());
+        //m_required_fields.emplace("gas_names", gas_names_layout, nondim, grid->name());
```

Comment:
any reason to keep this commented out code?

Reply:
Only because I was hoping to find a way to keep track of which gas names corresponded to specific indices in the gas concentrations array. Right now I'm just assuming they don't change. I removed the commented out code though, because commented out code is silly.